### PR TITLE
[ci-app] Allow Users to Provide `git` Metadata

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What and why?

We want to allow users to provide git metadata manually, so that there's a fallback when other approaches fail.

⚠️ **Important**: do not merge until datadog-ci-spec is updated.

### How?

* Read certain environment variables and pass them as span tags. These tags take precedence over metadata extracted by other methods. 

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

**Review by yoannmoinet — APPROVED** [review]

File: src/helpers/__tests__/user-provided-git.test.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      [GIT_COMMIT_MESSAGE]: 'DD_GIT_COMMIT_MESSAGE',
+      [GIT_COMMIT_COMMITTER_DATE]: 'DD_GIT_COMMIT_COMMITTER_NAME',
+      [GIT_COMMIT_COMMITTER_EMAIL]: 'DD_GIT_COMMIT_COMMITTER_EMAIL',
+      [GIT_COMMIT_COMMITTER_NAME]: 'DD_GIT_COMMIT_COMMITTER_DATE',
+      [GIT_COMMIT_AUTHOR_DATE]: 'DD_GIT_COMMIT_AUTHOR_DATE',
+      [GIT_COMMIT_AUTHOR_EMAIL]: 'DD_GIT_COMMIT_AUTHOR_EMAIL',
+      [GIT_COMMIT_AUTHOR_NAME]: 'DD_GIT_COMMIT_AUTHOR_NAME',
+    })
+  })
+  it('returns an empty object is no user git is defined', () => {
```
Comment (edrevo) [inline src/helpers/__tests__/user-provided-git.test.ts:61]:
```suggestion
  it('returns an empty object if no user git is defined', () => {
```

File: src/helpers/interfaces.ts [hunk 2]
```diff
@@ -9,8 +9,12 @@ import {
   CI_STAGE_NAME,
   CI_WORKSPACE_PATH,
   GIT_BRANCH,
+  GIT_COMMIT_AUTHOR_DATE,
   GIT_COMMIT_AUTHOR_EMAIL,
   GIT_COMMIT_AUTHOR_NAME,
+  GIT_COMMIT_COMMITTER_DATE,
+  GIT_COMMIT_COMMITTER_EMAIL,
+  GIT_COMMIT_COMMITTER_NAME,
```
Comment (edrevo) [inline src/helpers/interfaces.ts:17]: shouldn't we update getGitMetadata() to read these new tags from the .git folder?

**Review by m-rousse — APPROVED** [review]
LGTM infra-wise

File: src/helpers/interfaces.ts (hunk 2 above)
Comment (juan-fernandez) [reply src/helpers/interfaces.ts:17]: it does already, but you made me realise that we gain from adding an output type to `getGitMetadata`

File: src/helpers/__tests__/user-provided-git.test.ts (hunk 1 above)
Comment (juan-fernandez) [reply src/helpers/__tests__/user-provided-git.test.ts:61]: good catch

**Review by edrevo — APPROVED** [review]

**Pull request merged**